### PR TITLE
added pundits to policy and controller

### PR DESCRIPTION
**Pull request by TofuSupreme**
Deleted old PR. This is a newer version for Kondos only.

### CODE REVIEW

**Comment by spioupiou** [comment]
I think new and create in the kondo policy totally make sense, those routes are only accessible for providers :) For edit/update/destroy, I wonder if we should be more strict and make sure that the person editing or deleting the kondo is the creator of that kondo (something like current_user.id == kondo.user_id ?)

File: app/policies/kondo_policy.rb [hunk 1]
```diff
@@ -3,5 +3,25 @@ class Scope < Scope
     def resolve
       scope.all
     end
+
+    def new?
+      current_user.provider?
+    end
```
Comment (carlnoval) [inline app/policies/kondo_policy.rb:9]: This code is not needed, reason being is that inside the application_policy.rb the `new?` method calls the `create?` method from this file. Would be happy to explain more if its not clear :)

File: app/policies/kondo_policy.rb [hunk 2]
```diff
@@ -3,5 +3,25 @@ class Scope < Scope
...
+
+    def create?
+      current_user.provider?
+    end
+
+    def edit?
+      current_user.id == kondo.user_id
+    end
```
Comment (carlnoval) [inline app/policies/kondo_policy.rb:17]: This code is not needed, reason being is that inside the application_policy.rb the `edit?` method calls the `update?` method from this file. Would be happy to explain more if its not clear :)

**Comment by carlnoval** [comment]
All tests below are only via UI, not direct url

**Renter**
- [x] can access list of kondo services
- [x] can view a kondo
- [x] can book kondos
- [x] can't create kondos
- [x] can't can't edit kondos

**Provider**
- [x] can see all list of kondos that they own
- [x] can't see list of all kondos offered by all providers
- [ ] can create their kondo - not yet implemented
- [x] can view their kondo
- [ ] can edit their kondo - not yet implemented

On top of the above tests, I think we need to remove the conditional statements on the views to actually see pundit take effect. We'll have to then just fix additional issues that may come up.

**Comment by carlnoval** [comment]
Seems like a stray folder that needs to be removed
Screen Shot 2022-01-29 at 2 32 53

File: db/schema.rb [hunk 3]
```diff
@@ -56,6 +69,7 @@
     t.integer "price"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
+    t.integer "service_duration"
     t.index ["user_id"], name: "index_kondos_on_user_id"
   end
```
Comment (carlnoval) [inline db/schema.rb:75]: Won't really matter but I learned recently that there is no need for us to submit changes on this file. The reason being is that this file is fully dependent on the migrations so running `rails db:migrate` should generate the right `schema.rb` depending on how updated the repo's migrate folder.

**Pull request merged**